python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -469,107 +469,275 @@ def update_directions(db_manager: DatabaseManager) -> None:
-            .join(
-                StaticStops,
-                sa.and_(
-                    StaticStopTimes.stop_id == StaticStops.stop_id,
-                    StaticStopTimes.static_version_key
-                    == StaticStops.static_version_key,
-                ),
-            )
-            .where(
-                StaticRoutePatterns.static_version_key == version_key,
-                StaticRoutePatterns.route_pattern_typicality == 1,
-                StaticStopTimes.static_version_key == version_key,
-                StaticTrips.static_version_key == version_key,
-                StaticStops.static_version_key == version_key,
-            )
-            .subquery("static_sub")
+                    VehicleTrips.branch_route_id,
+                    VehicleTrips.trunk_route_id,
+                )
+                == static_canon.c.route_id,
+                VehicleTrips.static_version_key
+                == static_canon.c.static_version_key,
+                VehicleEvents.parent_station == static_canon.c.parent_station,
+            ),
         )
-
-        canonical_sub = (
-            sa.select(
-                VehicleEvents.pm_event_id,
-                static_sub.c.stop_sequence,
-            )
-            .select_from(VehicleEvents)
-            .join(
-                VehicleTrips,
-                VehicleEvents.pm_trip_id == VehicleTrips.pm_trip_id,
-            )
-            .join(
-                static_sub,
-                sa.and_(
-                    VehicleTrips.direction_id == static_sub.c.direction_id,
-                    sa.func.coalesce(
-                        VehicleTrips.branch_route_id,
-                        VehicleTrips.trunk_route_id,
-                    )
-                    == static_sub.c.route_id,
-                    VehicleEvents.parent_station == static_sub.c.parent_station,
-                ),
-            )
-            .where(
-                VehicleEvents.service_date == service_date,
-                VehicleTrips.static_version_key == version_key,
-            )
-            .subquery("canonical_sub")
+        .where(
+            VehicleEvents.service_date
+            == sa.func.any(
+                sa.func.array(
+                    sa.select(TempEventCompare.service_date)
+                    .distinct()
+                    .scalar_subquery()
+                )
+            ),
         )
+        .subquery("rt_canon")
+    )
 
-        update_query = (
-            sa.update(VehicleEvents.__table__)
-            .values(
-                canonical_stop_sequence=canonical_sub.c.stop_sequence,
-            )
-            .where(
-                VehicleEvents.pm_event_id == canonical_sub.c.pm_event_id,
-            )
+    update_rt_canon = (
+        sa.update(VehicleEvents.__table__)
+        .values(
+            canonical_stop_sequence=rt_canon.c.stop_sequence,
         )
+        .where(
+            VehicleEvents.pm_event_id == rt_canon.c.pm_event_id,
+        )
+    )
```

Comment:
so we're all the same as we were before here, right?

Reply:
Yup, nothing changed here.